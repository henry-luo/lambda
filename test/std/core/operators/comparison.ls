// Test: Comparison Operators
// Category: core/operators
// Type: positive
// Expected: 30

// Simple arithmetic comparison test (since < operator not supported)
let a = 10
let b = 20

// Test arithmetic instead of comparison
a + b
