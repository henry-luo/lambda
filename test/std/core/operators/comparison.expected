09:15:47 [NOTICE] Start transpiling test/std/core/operators/comparison.ls...
[script:list]
  [content:list[3]]
    no declare
    item:
    [let stam:null]
      declare:
      [assign expr:a:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [let stam:null]
      declare:
      [assign expr:b:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [binary expr +.3:int]
      [primary expr:int,const:1]
        [ident:a:int,const:1]
      [primary expr:int,const:1]
        [ident:b:int,const:1]
09:15:47 [NOTICE] compiling C code in 'test/std/core/operators/comparison.ls' to MIR
C2MIR init end           -- 68 usec
test/std/core/operators/comparison.ls:1:2: warning -- unknown pragma
    preprocessor tokens -- 2426, parse tokens -- 1458
  C2MIR preprocessor end    -- 502 usec
  C2MIR parser end          -- 625 usec
  C2MIR context checker end -- 709 usec
  C2MIR generator end       -- 814 usec
C2MIR compiler end                -- 852 usec
09:15:47 [NOTICE] Generating native code...
##### Script 'test/std/core/operators/comparison.ls' executed: #####
30
